### PR TITLE
Add iati.fetch.get_publishers

### PR DESCRIPTION
**Pull request by dalepotter**
Add function to get a list of publishers who are publishing to the IATI Registry.

### CODE REVIEW

File: iati/fetch/__init__.py [hunk 1]
```diff
@@ -97,3 +98,25 @@ def is_response_okay(request):
...
+
+    Args:
+        registry_api_endpoint (str): The API endpoint URL for obtaining a list of all organisations publishing to the IATI Registry.
+
+    Raises:
+        Exception: When a HTTP 200 response code is not received.
+
+    Returns:
+        list: List of Registry publisher IDs.
+    """
```
Comment (allthatilk) [inline iati/fetch/__init__.py:115]: Add Todo for replacing is_response_okay method.

File: iati/fetch/tests/test_dataset.py [hunk 2]
```diff
@@ -72,3 +73,39 @@ def test_get_dataset_correctly_encoded(self):
     def test_get_dataset_bad_url(self):
         """Given an invalid dataset URL, an Exception is raised."""
         pass
+
+    def test_get_publishers(self):
+        """Result of iati.fetch.get_publishers() will be a list and contain expected results."""
+        mock_publishers_response = pkg_resources.resource_stream(__name__, 'organization_list.json').read().decode('utf-8')
+        mock_publishers_url = 'http://test.com/publishers'
```
Comment (allthatilk) [inline iati/fetch/tests/test_dataset.py]: Have a look at changes to the TestDataset tests and extracting the mock URL to keep things DRY.

**Review by allthatilk — CHANGES_REQUESTED** [review]
A couple minor changes, main one for mock URL layout to keep it consistent with get_dataset

**Comment by dalepotter** [comment]
Thanks for the review - will look to fix these issues outside of project time. Hopefully tomorrow.

File: iati/fetch/__init__.py (hunk 1 above)
Comment (dalepotter) [reply iati/fetch/__init__.py:115]: On reflection I think it's ok to not include a todo here. 

There is already a todo in the `is_response_okay` docstring. Also, when the function is removed all usage of the function should be replaced anyway.

File: iati/fetch/tests/test_dataset.py (hunk 2 above)
Comment (dalepotter) [reply iati/fetch/tests/test_dataset.py]: I've made some changes in 655ba00 - is this what you mean?

File: iati/fetch/__init__.py (hunk 1 above)
Comment (allthatilk) [reply iati/fetch/__init__.py:115]: Cool beans!

**Review by allthatilk — APPROVED** [review]

**Pull request merged**